[grub2] grub2-mkconfig loads ext4 and brctl kernel modules

pmoravec opened the pull request:
Execute grub2-mkconfig only when both ext4 and brctl kernel modules
are loaded.

Resolves #821

Signed-off-by: Pavel Moravec <pmoravec@redhat.com>

bmr-cymru commented on sos/plugins/grub2.py:
```diff
@@ -33,10 +33,13 @@ def setup(self):
+        # This behaviour causes an unexpected configuration change for system.
+        # sosreport should aovid such situation.
+        if self.is_module_loaded("btrfs") and self.is_module_loaded("ext4"):
+            self.add_cmd_output("grub2-mkconfig")
```
I don't really like this as a solution; there will be many systems on RHEL7 that do not meet the btrfs|ext4 conditions and where we will lose diagnostic information as a result of this change.

I think we need to take this back to `grub2` and `os-prober`.

pmoravec commented:
@bmr-cymru Are there some comments to the PR or could it be merged, please?

bmr-cymru commented on sos/plugins/grub2.py:
```diff
@@ -33,10 +33,13 @@ def setup(self):
+        # possible unwanted loading of some kernel modules
+        env = {}
+        env['GRUB_DISABLE_OS_PROBER'] = 'true'
+        self.add_cmd_output("grub2-mkconfig", envchange=env)
```
This can't work: the previous patch adds a kwarg named `env_vars`; this patch is using `envchange`.

bmr-cymru commented on sos/plugins/__init__.py:
```diff
@@ -521,14 +521,15 @@ def add_copy_spec(self, copyspecs):
             self._log_info("added copyspec '%s'" % copy_paths)
```
Commit message is a tiny bit inaccurate: the passed environment is added to `os.environ`, rather than overwriting it entirely (but values from the argument do take precedence).

bmr-cymru commented on sos/plugins/__init__.py:
```diff
@@ -521,14 +521,15 @@ def add_copy_spec(self, copyspecs):
 
     def get_command_output(self, prog, timeout=300, stderr=True,
-                           chroot=True, runat=None):
+                           chroot=True, runat=None, env_vars=None):
```
Not sure I like `env_vars`.. `env` might be enough, or `environ`.

pmoravec commented:
The branch https://github.com/pmoravec/sos/tree/sos-pmoravec-grub2-loads-ext4-brctl-kernel-modules was:

1) rebased to current upstream (some conflicts resolved)
2) updated in the new commits (how I was so blind..)

The https://github.com/pmoravec/sos/tree/sos-pmoravec-grub2-loads-ext4-brctl-kernel-modules is now really 2 commits ahead sosreport:master and those 2 commits are the subject of this PR. No idea why github thinks this branch has conflicts..
